### PR TITLE
Build with distribution server

### PR DESCRIPTION
**Pull request by jrvollmer**
This pairs with https://github.com/badgerloop-software/engineering-data-distributor/pull/1.

---

**Changes**
- Build and run the chase-car-dashboard image alongside the engineering-data-distributor image in 2 configurations: individual and dev-no-datagen
  - Currently, only "individual" is working due to issues with the containers' networks, but I should have "dev-no-datagen" working tonight with a similar solution to what was used for the "individual" config
  - With both of these configurations, the engineering-data-distributor image's sc1-data-format submodule will automatically inherit the same commit as that of the chase-car-dashboard image's submodule so that there aren't any conflicts with the byte array
  - **@EUdds and @MCLiii, what are your thoughts on making "individual" the default configuration instead of "competition"?** Since we're designing the dashboard to run for competition, I thought it made sense initially. However, it can't be run without someone who knows what they're doing to set up and run the engineering data distributor on another device, which makes the configuration unsuitable for general use. **I'm considering changing the default configuration to "individual" since it enables communication with the solar car via a cellular connection and the radios (same as the "competition" config) and requires no setup unless you want to use the radios. This _should_ make it more suitable for general use and hardware testing as well.**
- Build and run the chase-car-dashboard image on its own in 2 configurations: competition (default) and dev
  - "dev" has been tested and works
  - I have to test "competition," but I don't foresee it having any glaring issues (_knock on wood_) since it's the same as the image's original implementation, just with new implications regarding the device it's connecting to

---

**TODOs (@jrvollmer)**
- [X] Update `windows-pull-run.bat` to reflect the changes made to `linux-pull-run.sh`
  - [x] In the usage or README, add a note about ensuring that your firewall is off or has a rule to allow connections from 192.168.1.0/24 on your computer to 192.168.1.15 externally
- [x] Update the README with instructions about the network setup for each configuration
  - [x] I might just make a table with a summary of this information in the README for brevity and the sake of time :worried:. Detailed setup instructions will be added in Confluence
  - NOTE: Just added details in usage for each script and pointed to the usage in the README

### CODE REVIEW

File: Backend/src/constants.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
-    "CAR_PORT": 4003,
-    "TEST_ADDRESS": "localhost",
-    "PI_ADDRESS": "192.168.1.15",
+    "TYP_INCOMING_DATA_PORT": 4003,
```
Comment (EUdds) [inline Backend/src/constants.json]: TCP?

**Review by EUdds — CHANGES_REQUESTED** [review]

File: Backend/src/constants.json (hunk 1 above)
Comment (jrvollmer) [reply Backend/src/constants.json]: It was supposed to be "TYP" for "TYPICAL." I just didn't want it to be too long, but I can think of something better

Comment (EUdds) [reply Backend/src/constants.json]: Usually NOM for NOMINAL or NORMAL

Comment (jrvollmer) [reply Backend/src/constants.json]: NORM it is

File: linux-pull-run.sh [hunk 2]
```diff
@@ -53,14 +93,65 @@ while [[ "$1" =~ ^- && ! "$1" == "--" ]]; do case $1 in
...
+		;;
+	-c | --config )
+		# Make sure a configuration was not already specified
+		_enforce_one_configuration
+
+		# Get the configuration specified after the -c or --config flag
+		config=$2
+
+		# Check that a valid configuration was provided
+		if [[ "$config" != @(competition|dev|individual) ]]; then
```
Comment (MCLiii) [inline linux-pull-run.sh]:
```suggestion
		if [[ ! "$config" =~ ^(competition|dev|individual)$ ]]; then
```
I'm getting a syntax error here running on my Mac, changing to this seems to work

File: linux-pull-run.sh [hunk 3]
```diff
@@ -53,14 +93,65 @@ while [[ "$1" =~ ^- && ! "$1" == "--" ]]; do case $1 in
...
+		;;
+	--config=* )
+		# Make sure a configuration was not already specified
+		_enforce_one_configuration
+
+		# Get the characters immediately after the equals sign in the CLA
+		config="${1##--config=}"
+
+		# Check that a valid configuration was provided
+		if [[ "$config" != @(competition|dev|individual) ]]; then
```
Comment (MCLiii) [inline linux-pull-run.sh]:
```suggestion
		if [[ ! "$config" =~ ^(competition|dev|individual)$ ]]; then
```
Same here

**Review by MCLiii — COMMENTED** [review]
Minor issues with linux pull script otherwise it looks good

File: linux-pull-run.sh (hunk 2 above)
Comment (jrvollmer) [reply linux-pull-run.sh]: Fixed

File: linux-pull-run.sh (hunk 3 above)
Comment (jrvollmer) [reply linux-pull-run.sh]: Fixed

**Review by MCLiii — APPROVED** [review]
LGTM!

**Pull request merged**